src/main/java/com/cosmeticos/service/OrderService.java

```diff
@@ -376,24 +444,192 @@ private void validateScheduledAndsendPaymentRequest(Order persistenOrder) throws
         //PEGAMOS A DATA DE INICIO DO AGENDAMENTO DO PEDIDO
         Date scheduleDateStart = persistenOrder.getScheduleId().getScheduleStart();
 
-        //ATRIBUIMOS A DATA DO AGENDAMENTO DO PEDIDO AO CALENDARIO
-        c.setTime(scheduleDateStart);
+
+		//--- CONFIGURACOES PARA DEFINIR A DATA QUE DEVEMOS INICIAR AS COBRANCAS ---//
+		//ATRIBUIMOS A DATA DO AGENDAMENTO DO PEDIDO AO CALENDARIO
+		c.setTime(scheduleDateStart);
 
         //VOLTAMOS N DIAS, DEFINIDO EM PROPRIEDADES, NO CALENDARIO BASEADO NA DATA DO AGENDAMENTO
         c.add(Calendar.DATE, -daysToStart);
 
-        //DATA DO AGENDAMENTO MENOS N DIAS NO FORMADO DATE. OU SEJA, A DATA QUE DEVE INICIAR AS TENTAVIDAS DE PAGAMENTO
-        Date dateToStartPayment = c.getTime();
-        //TODO - DEVERIAMOS COBRAR SOMENTE SE FOR ATE A DATA DE AGENDAMENTO? POIS CORREMOS O RISCO DE COBRAR ALGO BEM ANTIGO
-        //SE A DATA ATUAL FOR POSTERIOR A DATA QUE DEVE INICIAR AS TENTATIVAS DE RESERVA DO PAGAMENTO, ENVIAMOS PARA PAGAMENTO
-        if (now.after(dateToStartPayment)) {
-            sendPaymentRequest(persistenOrder);
-        }
-    }
+		//DATA DO AGENDAMENTO MENOS N DIAS NO FORMADO DATE. OU SEJA, A DATA QUE DEVE INICIAR AS TENTAVIDAS DE PAGAMENTO
+		Date dateToStartPayment = c.getTime();
+
+
+		//--- CONFIGURACOES PARA DEFINIR A DATA QUE DEVEMOS INICIAR AS NOTIFICACOES AO CLIENTE CASO FALHE ---//
+		//ATRIBUIMOS A DATA DO AGENDAMENTO DO PEDIDO AO CALENDARIO
+		c.setTime(scheduleDateStart);
+
+		//VOLTAMOS N DIAS, DEFINIDO EM PROPRIEDADES, NO CALENDARIO BASEADO NA DATA DO AGENDAMENTO
+		c.add(Calendar.DATE, -daysBeforeStart);
+
+		//UM DIA ANTES PARA NOTIFICAR AO CLIENTE SE DER ERRO NA RESERVA NO CARTAO E SUGERIR TROCAR PARA DINHEIRO
+		Date dateToStartNotification = c.getTime();
+
+
+        //TODO - VERIFICAR SE E UM DIA ANTES E TENTAR ENVIAR REQUEST, SE DER ERRO, LOGAR PARA DEPOIS NOTIFICAR NO APP
+		if(sdf.format(now).equals(sdf.format(dateToStartNotification))) {
+			//TODO - URGENTE: VERIFICAR MELHORIA, POIS O ERRO AQUI PODE SER DE REDE E ETC, NAO SO DE LIMITE DO CARTAO
+			if(!sendPaymentRequest(persistenOrder)) {
+				//AQUI O GARRY DISSE QUE TROCARIAMOS, POSTERIORMENTE, PARA ALGO QUE IRA GERAR O POPUP NA TELA DO CLIENTE
+				log.error("Erro ao efetuar a reserva do pagamento, sugerimos que troque o pagamento para dinheiro");
```

Comment:
Aqui teremos que lançar uma exceção com um type associado ao http status coerente com o erro.

Reply:
Vish, e qual seria o status???

402 Payment Required
Reserved for future use. The original intention was that this code might be used as part of some form of digital cash or micropayment scheme, as proposed for example by GNU Taler[34], but that has not yet happened, and this code is not usually used. Google Developers API uses this status if a particular developer has exceeded the daily limit on requests.[35]

403 Forbidden
The request was valid, but the server is refusing action. The user might not have the necessary permissions for a resource, or may need an account of some sort.

424 Failed Dependency (WebDAV; RFC 4918)
The request failed due to failure of a previous request (e.g., a PROPPATCH).[15]

449 Retry With
The server cannot honour the request because the user has not provided the required information.[78]

https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

Reply:
Aborta... pensei que essa instrução tava dentro do fluxo de request, mas é de um fluxo schedulado. Nao tem pra quem retornar httpstatus. Ignora meu comentario.